### PR TITLE
Issue07 print records

### PR DESCRIPTION
**Pull request by jpalcala**
* Added printing functionality

### CODE REVIEW

File: src/Grid/RemoteDataSource.js [hunk 1]
```diff
@@ -31,6 +31,37 @@ class RemoteDataSource {
     this._doRequest(rowsPerPage, page);
   }
 
+  getAllRecords = (count, searchText) => new Promise((resolve, reject) => {
```
Comment (geoperez) [inline src/Grid/RemoteDataSource.js]: Nop, why don't reuse `doRequest`?

File: src/Grid/Grid.js [hunk 2]
```diff
@@ -45,6 +49,35 @@ class Grid extends React.Component {
     this.state.dataSource.search(this.state.rowsPerPage, this.state.page, text);
   }
 
+  printTable = () => {
+    const { dataSource, filteredRecordCount, searchText } = this.state;
+    dataSource.getAllRecords(filteredRecordCount, searchText)
+      .then(rows => {
+        const popup = window.open('about:blank', 'Print', 'location=0,height=500,width=800');
+        popup.document.write('<link rel="stylesheet" href="//cdn.jsdelivr.net/bootstrap/latest/css/bootstrap.min.css" />');
```
Comment (geoperez) [inline src/Grid/Grid.js]: Why the bootstrap template?

File: src/Grid/RemoteDataSource.js (hunk 1 above)
Comment (jpalcala) [reply src/Grid/RemoteDataSource.js]: Because I don't want to modify the current table items.

Comment (geoperez) [reply src/Grid/RemoteDataSource.js]: Invalid reason, common. Reuse some code.

Comment (jpalcala) [reply src/Grid/RemoteDataSource.js]: Forgive me father for I have sinned

Comment (ArCiGoUnoSquare) [reply src/Grid/RemoteDataSource.js]: 😱

File: src/Grid/Grid.js [hunk 3]
```diff
@@ -51,16 +51,17 @@ class Grid extends React.Component {
...
         const popup = window.open('about:blank', 'Print', 'location=0,height=500,width=800');
         popup.document.write('<link rel="stylesheet" href="//cdn.jsdelivr.net/bootstrap/latest/css/bootstrap.min.css" />');
         const tableHtml = `<table class="table table-bordered table-striped"><thead><tr>${
           dataSource.columns
             .filter(c => c.Visible)
             .reduce((prev, el) => `${prev}<th>${el.Label || el.Name}</th>`, '')
         }</tr></thead><tbody>${ 
-          rows.map(row => {
+          payload.map(row => {
             if(typeof(row) === 'object'){
```
Comment (geoperez) [inline src/Grid/Grid.js]: maybe use `row instanceof Object`

File: src/Grid/RemoteDataSource.js [hunk 4]
```diff
@@ -89,39 +94,13 @@ class RemoteDataSource {
   }
   
   _doRequest(rowsPerPage, page, searchText) {
```
Comment (kadosh) [inline src/Grid/RemoteDataSource.js]: This method shouldn't be "_doRequest" anymore, since the actual method doing the request is "getAllRecords"

**Pull request merged**